src/dev.ts

```diff
@@ -0,0 +1,279 @@
+    await this.startInteractiveMode()
+  }
+
+  private async startServer () {
+    console.log(colors.blue('🚀 Starting development server with hot reload...'))
+
+    // Start the server in the background using our robust serve infrastructure
+    const serverArgs = [
+      '--dp', String(this.options.dashboardPort || 3000),
+      '--port', String(this.options.port || 8000),
+      '--db-type', 'mem', // Use memory for faster dev iterations
+      this.projectRoot
+    ]
+
+    try {
+      // Start the serve function in the background
+      // Don't await it here as it runs indefinitely
+      this.serverProcess = serve(serverArgs).catch(error => {
+        console.error(colors.red('❌ Server error:'), error)
+        process.exit(1)
+      })
+
+      // Give the server a moment to start
+      await new Promise(resolve => setTimeout(resolve, 2000))
+
+      console.log(colors.green('✅ Development server started'))
+      console.log(colors.gray(`   Dashboard: http://localhost:${this.options.dashboardPort || 3000}`))
+      console.log(colors.gray(`   Application: http://localhost:${this.options.port || 8000}`))
+    } catch (error) {
+      console.error(colors.red('❌ Failed to start development server:'), error)
+      throw error
+    }
+  }
+
+  private printWelcomeBanner () {
+    console.log(colors.cyan('╔══════════════════════════════════════════════════════════════╗'))
+    console.log(colors.cyan('║') + colors.bold(colors.white('               🧪 CHEL LIVE DEVELOPMENT                     ')) + colors.cyan('║'))
+    console.log(colors.cyan('║') + colors.white('           Live-testing with Hot Reload & Interaction        ') + colors.cyan('║'))
+    console.log(colors.cyan('╚══════════════════════════════════════════════════════════════╝'))
+    console.log()
+    console.log(colors.gray('Similar to grunt dev - provides live-testing environment'))
+    console.log(colors.gray('with automatic contract redeployment and interactive testing.'))
+    console.log()
+  }
+
+  private async startContractWatching () {
+    console.log(colors.blue('👀 Setting up contract file watching...'))
+    const contractsDir = join(this.projectRoot, 'contracts')
+
+    if (!existsSync(contractsDir)) {
+      console.log(colors.yellow('⚠️  No contracts directory found, creating one...'))
+      await mkdir(contractsDir, { recursive: true })
+      return
+    }
+
+    try {
+      // Watch for changes in contract files
+      const watcher = watch(contractsDir, { recursive: true }, (eventType, filename) => {
+        if (filename && (filename.endsWith('.js') || filename.endsWith('.json'))) {
+          this.handleContractChange(eventType, filename)
+        }
+      })
+
+      this.watchers.push(watcher)
+
+      // Count initial contracts
+      const entries = await readdir(contractsDir, { recursive: true })
+      const contractFiles = (entries as unknown as string[]).filter((f) =>
+        typeof f === 'string' && f.endsWith('.js')
```

Comment:
This business with the `.js` extension isn't the best way for implementing this, since there's no requirement for contracts to bear any specific extension (or even an extension at all). However, we do have files with a `.manifest.json` extension, which you can reasonably expect to be manifests. From there, you can read the manifest contents to find out the file paths.

Reply:
The essence of this comment stands, but it was primarily meant for `pin`. Here, it's not too bad to do this, but it'd still be good to do it without having to rely on the extension.